lib/cli/ui/terminal.rb

```diff
@@ -7,30 +7,52 @@ module Terminal
       DEFAULT_WIDTH = 80
       DEFAULT_HEIGHT = 24
 
+      @console = begin
```

Comment:
The code reads a bit weirdly with all those begins.

To simplify and reduce the number of cached variables, how about:

```rb
def self.width
  winsize[1]
end

def self.height
  winsize[0]
end

def self.winsize
  @winsize ||= begin
    winsize_trap
    console = IO.respond_to?(:console) && IO.console

    if (winsize = console&.winsize) && winsize.none?(&:zero?)
      @winsize = winsize
    else
      [DEFAULT_HEIGHT, DEFAULT_WIDTH]
    end
  rescue Errno::EIO
    [DEFAULT_HEIGHT, DEFAULT_WIDTH]
  end
end

def self.winsize_trap
  @winsize_trap ||= Signal.trap('WINCH') do
    @winsize = nil
  end
end
```

Note how I changed the trap setup to be done lazily so that we only ever do it if requesting winsize